federation/pkg/dnsprovider/providers/google/clouddns/rrsets.go

```diff
@@ -30,31 +32,46 @@ type ResourceRecordSets struct {
 	impl interfaces.ResourceRecordSetsService
 }
 
+// List returns a list of resource records in the given project and
+// managed zone.
+// !!CAUTION!! Your memory might explode if you have a huge number of
+// records in your managed zone.
```

Comment:
What is this comment based upon? We've seen a lot of recordsets in the jenkins test projects that accumulate orphaned rrsets over time but even with multiple thousands, it shouldn't be enough to OOM the controller.